### ISSUE TITLE
[Feature]: Video Scrubbing with Preview Thumbnails

### ISSUE THREAD

**Issue by coofzilla** [TheWidlarzGroup/react-native-video#4094]
### Description

Adds video scrubbing with thumbnail previews to enhance navigation. 

![image](https://github.com/user-attachments/assets/5e904177-e8bd-4913-87f5-3da7f2c0c803)

### Why it is needed ?

Improves user experience by providing visual cues during scrubbing.

### Possible implementation

For `ios` to support `hls` I believe it would involve implementing https://developer.apple.com/documentation/http-live-streaming/hls-authoring-specification-for-apple-devices#Trick-Play

**Comment by moskalakamil** [comment]
Hi @coofzilla, thanks for sharing an idea! Right now it's possible to show preview on video while seeking thanks to Plugins System. You can read more about how to integrate this feature on this blog https://medium.com/@freeboub/react-native-video-plugin-thumbnail-view-5a4a227299b8. And you see the demo here https://github.com/TheWidlarzGroup/react-native-video/pull/3943

**Comment by coofzilla** [comment]
Is there a roadmap for when `plugins` are going to be "stable"? Also, what is the intention on the possibilities, from what I read this is mainly for `analytics` features? but, perhaps I'm misinterpreting this?  

I was imagining something where the community can create plugins for `niche` features and then this way add them as we need it. Is that the direction these plugins are going?

**Comment by freeboub** [comment]
@coofzilla the api is marked as unstable because we don't have much feedback on the implementation. So I hope it will not change. Of course, you can use it and also use it for a thumbnail preview. I wrote a POC with an explanation here : https://medium.com/@freeboub/react-native-video-plugin-thumbnail-view-5a4a227299b8

I hope it helps !

**Comment by freeboub** [comment]
@coofzilla I am also thinking of creating a plug-in to synchronize 2 players (linked to your other feature request). I can elaborate more if you want details on my idea, but unfortunately, I don't have time to implement it ...

**Comment by coofzilla** [comment]
> @coofzilla I am also thinking of creating a plug-in to synchronize 2 players (linked to your other feature request). I can elaborate more if you want details on my idea, but unfortunately, I don't have time to implement it ...

yeah I'd love to hear about it! I haven't had a chance to dive into plugins for `react-native-video` ; so, I'm curious how they work and whats possible with them.

**Comment by matgren** [comment]
@coofzilla have you seen https://x.com/krzysztof_moch/status/1867161660127084759 ? perhaps this is what you were looking for.
